pkg/controller/storagecluster/storagecluster.go

```diff
@@ -725,12 +725,11 @@ func (c *Controller) createPodTemplateForNodeGroup(
 	remainingNodes map[string]*v1.Node,
 	hash string,
 ) error {
-	podTemplate, err := c.createPodTemplate(cluster, nodeGroup[0], hash)
-	if err != nil {
-		return err
-	}
-
 	for _, node := range nodeGroup {
+		podTemplate, err := c.createPodTemplate(cluster, node, hash)
```

Comment:
Thanks for catching this. Can you also update the comment to this function and remove `It creates a single pod template... ` line?

Reply:
fixed